src/routes/Add.jsx

```diff
@@ -1,13 +1,129 @@
-import { Card, CardHeader } from '@nextui-org/react';
+import { useState } from 'react';
+import {
+  Card,
+  CardBody,
+  Input,
+  CardHeader,
+  Divider,
+  Select,
+  SelectItem,
+  Textarea,
+  Button
+} from '@nextui-org/react';
+import { serverTimestamp } from 'firebase/firestore';
+import LoadingCard from '../pages/Dictionary/components/LoadingCard';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { unpackTermsQuery } from '../utils/unpackQuery';
+import getTermsQuery from '../utils/getTermsQuery';
+import getEntriesMutation from '../utils/getEntriesMutation';
 
 const Add = () => {
+  const [termSelectBox, setTermSelectBox] = useState('');
+  const [definitionTextArea, setDefinitionTextArea] = useState('');
+  const [exampleInput, setExampleInput] = useState('');
+  const [tagInput, setTagInput] = useState([]);
+  const navigate = useNavigate();
+
+  const [searchParams] = useSearchParams();
```

Comment:
i think you don't need to get searchParams in Add component. you just need to fetch all terms list like this: `const termsQuery = getTermsQuery();`

Reply:
done